ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/materialization/ks/KsMaterializedTableTest.java

```diff
@@ -105,9 +118,9 @@ public void shouldThrowNPEs() {
   }
 
   @Test
-  public void shouldThrowIfGettingStateStoreFails() {
+  public void shouldThrowIfQueryFails() {
     // Given:
-    when(stateStore.store(any(), anyInt())).thenThrow(new MaterializationTimeOutException("Boom"));
+    when(stateStore.getKafkaStreams().query(any())).thenThrow(new MaterializationTimeOutException("Boom"));
```

Comment:
We might also return a failed QueryResult. Should we add a test for that?

Reply:
We definitely should! :)